macro-ipld-derive/src/lib.rs

```diff
@@ -0,0 +1,35 @@
+use proc_macro::TokenStream;
+use quote::quote;
+use syn::{parse_macro_input, DeriveInput};
+
+#[proc_macro_derive(SerdeIpld)]
+pub fn derive_serde_ipld(input: TokenStream) -> TokenStream {
+    let ast = parse_macro_input!(input as DeriveInput);
+
+    let name = &ast.ident;
+
+    let expanded = quote! {
+            // impl MacroTrait for #name // should we make a trait?
+            impl #name {
```

Comment:
Yes we should have a trait here that way we have some context unto which to attach the methods. Having a `to_json` method on a struct is not enough context to know that we are dealing with dag-json specifically. 

Whatever we name the trait should be the name of the derive macro instead of `SerdeIpld`. Then its clear we are deriving the behavior define in that trait.